### PR TITLE
[WIP] musl-fts: add package at 1.2.7

### PR DESCRIPTION
**Pull request by FriendlyNeighborhoodShane**
Hi! Newbie to porting packages here.

Upstream (https://github.com/pullmoll/musl-fts) does not seem to provide pre-autotool'd releases. Any way we can remove automake and libtool as deps? Would including the products of `bootstrap.sh` into the repo be safe?

If this is accomplished then we would be able to drop the gigantic patch we have for `nnn` as well as enable other software, notably even MAC systems like SELinux.

---

## Description of package

Vendoring/port of glibc `fts(3)` library for musl.

## New package

- [X] Latest upstream version.
- [X] I agree to maintain this package.
- [X] I agree to notify the @kiss-community maintainers if I can no longer maintain this package.

### CODE REVIEW

**Comment by FriendlyNeighborhoodShane** [comment]
Ugh. I managed to create an archive of the build products but it's now complaining about missing a local copy of autoconf because I changed the timestamps and stuff on those products.

**Comment by FriendlyNeighborhoodShane** [comment]
Here's me doing the maintainer's job lmao https://github.com/FriendlyNeighborhoodShane/community/tree/ftsbin

For the record, I'm fully aware of the insanity of this solution and don't really expect this route to be taken. As such, not adding this to the PR. Hate autotools, hate having to harcode data into a repository that is literally 50x the size of the relevant code. Also hate the extra point-of-trust this makes me.

But it does work on my setup now, even after autotools uninstalled.

**Comment by dilyn-corner** [comment]
If you want to generate those files without having to appeal to `bootstrap.sh`, I would probably suggest simply running it yourself locally and hosting that tarball on GitHub. I'd probably just fork the upstream repository which includes the files that come out of `bootstrap.sh`, and create release tags in sync with them (if it ever happens again), just to improve the 'chain of trust' that you haven't tainted or modified the sources. Then you can just use that tarball (as it's basically equivalent, except for a single additional commit). Otherwise, I'll leave some comments on this.

File: community/libfts/depends [hunk 1]
```diff
@@ -0,0 +1,3 @@
+automake  make
+libtool   make
+musl
```
Comment (dilyn-corner) [inline community/libfts/depends:3]: `musl` is an assumed dependency and can be left out.

File: community/libfts/build [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh -e
+
+./bootstrap.sh
+./configure \
+    --prefix=/usr
+
+make
+make \
+    DESTDIR="$1" \
+    install
```
Comment (dilyn-corner) [inline community/libfts/build:10]: `make \DESTDIR="$1" \install` can be on a single line (optionally forgoing `DESTDIR="$1"`).

**Comment by FriendlyNeighborhoodShane** [comment]
Made the relevant changes.

Rebased no-autotools commit: https://github.com/FriendlyNeighborhoodShane/community/commit/5d5c8057bc6af67e75db475f06fe1360a470c7e5

**Comment by FriendlyNeighborhoodShane** [comment]
I'd _certainly_ like to not have to keep autotools around just to use `nnn`. But I'm definitely not comfortable with an anonymous nobody internet account I have being a 'trusted' upstream for the package. Making my own tarball release on a fork does actually 'formalize' and make it a little more transparent though, compared to this harcoding build products in the repo. But I'd prefer it not involve me at all.

Might try more to see if something else can be done.

Searching around made me stumble on the musl wiki's entry on fts, which claimed that the glibc fts interface was massively broken and anybody using it properly included its own in-tree copy anyway. They're... not wrong. GNU Coreutils certainly agrees: https://github.com/coreutils/gnulib/blob/master/lib/fts_.h

Wonder why nnn (upstream) doesn't just include its own in-tree copy for it.

**Comment by dilyn-corner** [comment]
It's probably just because Jarun didn't look closely into why musl doesn't provide an fts implementation, and just saw the benefits in his testing https://github.com/jarun/nnn/commit/284a3c4866b77f0d452d91aaf026068dd1e44cb1. 

It's possible they could be convinced to do their own fts for `nnn`, but that's perhaps more work than they want; especially considering `musl-fts` exists, and they have no (explicit) qualms with it; I don't know how familiar they are with musl in the first place though. 

As for forking and hosting a ready-to-go source: because `musl-fts` hasn't seen any commits in YEARS, there's very little burden on you -- Pullmoll's been inactive for half a year in general on GitHub, so it's unclear whether there would ever be more development on it. It's arguably the most transparent, least sketch, and most in-line with (the previously existing) guidestones than merely bundling a tarball (which should be hosted and not included in the repositories anyways).

**Comment by FriendlyNeighborhoodShane** [comment]
> It's possible they could be convinced to do their own fts for `nnn`, but that's perhaps more work than they want; especially considering `musl-fts` exists, and they have no (explicit) qualms with it; I don't know how familiar they are with musl in the first place though.

True.

> As for forking and hosting a ready-to-go source: because `musl-fts` hasn't seen any commits in YEARS, there's very little burden on you -- Pullmoll's been inactive for half a year in general on GitHub, so it's unclear whether there would ever be more development on it. It's arguably the most transparent, least sketch, and most in-line with (the previously existing) guidestones than merely bundling a tarball (which should be hosted and not included in the repositories anyways).

It's not as much that I'm worried about a maintainance burden (I also doubt that it's going to see much changes soon. In fact, as the musl wiki mentions, there were even talks about glibc deprecating it), but I'm still a third-party that's hosting what are essentially build artifacts. I'd prefer my involvement in a chain of trust between the original source of the code and the user to be as small as possible, out of principles.

Come to think of it, can't we "inline" musl-fts into nnn's build, and just have a tiny patch for the makefile and includes? IIRC several packages in kiss used this approach. BRB.

> But if nobody adopts `nnn` soon, I'm dropping it - #197 :)

Big YIKES. Ah, I guess if I can make this work...

**Comment by FriendlyNeighborhoodShane** [comment]
Closed, because I have no interest in fts outside of nnn.

Inlining musl-fts into nnn build succeeded, see results in #683

**Pull request closed without merging**